x-pack/plugins/uptime/server/lib/requests/get_monitor_states.ts

```diff
@@ -68,11 +69,113 @@ export const getMonitorStates: UMElasticsearchQueryFn<
     statusFilter
   );
 
-  const page = await fetchPage(queryContext);
+  const size = Math.min(queryContext.size, QUERY.DEFAULT_AGGS_CAP);
```

Comment:
i am not sure what's the purpose of this statement? pageSize will never exceeds 100, so it will always be queryContext.size, why we need this?

Reply:
It's just extra enforcement to prevent bad queries. I suppose it isn't needed however.

Reply:
Do you mind if we leave it in, given that it's (slightly) beneficial